### DISCUSSION TITLE
how to write from a KubernetesPodOperator into Xcom?

### DISCUSSION THREAD

**Discussion by davidbelhamou** [apache/airflow#26843] [Q&A]
Hi,

I have a DAG that use KubernetesPodOperator, with a docker image i created.
this image contain a specific python file (this python file is using papermill to run notebook).
and in the cmds param from the operator i put cmds=python /directory/python_code.py.
so the pod created by the KubernetesPodOperator have to run the python_code.py file.
I want that this file could return a value and push this same value to xcoms for the next task.
How can i do it?

**Comment by Taragolis** [comment]
https://airflow.apache.org/docs/apache-airflow-providers-cncf-kubernetes/stable/operators.html#how-does-xcom-work

**Reply by davidbelhamou** [reply]
This way you push to Xcoms from inside the DAG code.
I want to push to Xcoms from the python file (python_code.py)

**Reply by Taragolis** [reply]
1. You need explicit set `do_xcom_push=True` for your task which use `KubernetesPodOperator`
2. Your python code `python_code.py` should save result into the `/airflow/xcom/return.json` in valid [JSON format](https://datatracker.ietf.org/doc/html/rfc7159.html) inside the pod
3. Profit

**Reply by davidbelhamou** [reply]
thank you.
Now i have another problem with the sidecar container.
I'm getting an error about the cpu of the sidecar container.
I saw that there is a solution to this bug but i haven't figure out what to do.
do you have an idea?
[https://github.com/apache/airflow/pull/16851](url)
